diff -ur LTK-1.0.0.5.orig/LTKC/Library/ltkc_platform.h LTK-1.0.0.5/LTKC/Library/ltkc_platform.h
--- LTK-1.0.0.5.orig/LTKC/Library/ltkc_platform.h	2008-01-29 17:56:24.000000000 -0700
+++ LTK-1.0.0.5/LTKC/Library/ltkc_platform.h	2009-03-05 15:32:59.000000000 -0700
@@ -28,8 +28,13 @@
 #include <stdlib.h>         /* malloc() */
 #include <string.h>         /* memcpy() */
 
+#ifndef FALSE
 #define FALSE       0
+#endif
+
+#ifndef TRUE
 #define TRUE        1
+#endif
 
 /*
  * Typedefs of simple types.
@@ -39,8 +44,6 @@
  * good to go.
  */
 
-#ifdef _STDINT_H
-
 typedef uint8_t                 llrp_u8_t;
 typedef int8_t                  llrp_s8_t;
 typedef uint16_t                llrp_u16_t;
@@ -55,4 +58,3 @@
 typedef int                     llrp_bool_t;
 typedef uint8_t                 llrp_byte_t;
 
-#endif /* _STDINT_H */
diff -ur LTK-1.0.0.5.orig/LTKC/Library/ltkc_xmltextencode.c LTK-1.0.0.5/LTKC/Library/ltkc_xmltextencode.c
--- LTK-1.0.0.5.orig/LTKC/Library/ltkc_xmltextencode.c	2008-10-17 11:51:50.000000000 -0600
+++ LTK-1.0.0.5/LTKC/Library/ltkc_xmltextencode.c	2009-03-05 15:31:44.000000000 -0700
@@ -1498,7 +1498,7 @@
         return;
     }
 
-    strcpy(&pEncoder->pBuffer[pEncoder->iNext], aHoldBuf);
+    strcpy((char*)&pEncoder->pBuffer[pEncoder->iNext], aHoldBuf);
 
     pEncoder->iNext += nHoldBuf;
 }
@@ -1542,7 +1542,7 @@
     /*
      * Construct an XML encoder
      */
-    pXMLEncoder = LLRP_XMLTextEncoder_construct(pBuffer, nBuffer);
+    pXMLEncoder = LLRP_XMLTextEncoder_construct((unsigned char *)pBuffer, nBuffer);
     if(NULL == pXMLEncoder)
     {
         strcpy(pBuffer, "ERROR: XMLTextEncoder_construct failed\n");
diff -ur LTK-1.0.0.5.orig/LTKC/Tests/dx101.c LTK-1.0.0.5/LTKC/Tests/dx101.c
--- LTK-1.0.0.5.orig/LTKC/Tests/dx101.c	2008-01-29 17:56:24.000000000 -0700
+++ LTK-1.0.0.5/LTKC/Tests/dx101.c	2009-03-05 15:31:44.000000000 -0700
@@ -287,7 +287,7 @@
             char                    aBuf[100*1024];
             LLRP_tSXMLTextEncoder * pEncoder;
 
-            pEncoder = LLRP_XMLTextEncoder_construct(aBuf, sizeof aBuf);
+            pEncoder = LLRP_XMLTextEncoder_construct((unsigned char *)aBuf, sizeof aBuf);
             LLRP_Encoder_encodeElement(&pEncoder->encoderHdr,
                     &pMessage->elementHdr);
             if(!pEncoder->bOverflow)
diff -ur LTK-1.0.0.5.orig/LTKC/Tests/dx201.c LTK-1.0.0.5/LTKC/Tests/dx201.c
--- LTK-1.0.0.5.orig/LTKC/Tests/dx201.c	2008-01-29 17:56:24.000000000 -0700
+++ LTK-1.0.0.5/LTKC/Tests/dx201.c	2009-03-05 15:31:44.000000000 -0700
@@ -259,7 +259,7 @@
         char *              pMemHiwatAtEnd = (char*)sbrk(0);
 
         printf("INFO: Needed %d bytes of heap\n",
-            pMemHiwatAtEnd - pMemHiwatAtStart);
+            (int)(pMemHiwatAtEnd - pMemHiwatAtStart));
     }
 #endif
 
